File: 1510-find-lucky-integer-in-an-array/find-lucky-integer-in-an-array.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var findLucky = function(arr) {
    const map = new Map();
     
        arr.forEach(num => {
            map.set(num, (map.get(num) || 0) + 1);
        });

        let max = -1;
        for (const [key, value] of map.entries()) {
            if (key === value) {
                max = Math.max(max, key);
            }
        }

        return max;
    
};

File: 1510-find-lucky-integer-in-an-array/find-lucky-integer-in-an-array.java
class Solution {
    public int findLucky(int[] arr) {
        int n = arr.length;
        HashMap<Integer,Integer> map = new HashMap<>();
        for(int i=0;i<n;i++){
            map.put(arr[i], map.getOrDefault(arr[i], 0) + 1);

        }
        int max = -1;
        for (Integer key : map.keySet()) {
            if (key.equals(map.get(key))) {
                max = Math.max(max, key);
            }
        }
        return max;
        
    }
}